### PR TITLE
feat(other): add documentation check to CI

### LINKED ISSUE #6816: 🔧 [Refactor] docs.ocelot.social - ensure vuepress does not break
## 🔧 Refactor ticket

The new doc.socelot.social based on vuepress is an external page.
The content comes from this repository.

Ensure that the docs page is always working correctly and does not accidentally break.

Depending on how the actual implementation is this might change. But expecting an index called `SUMMARY.md` and other `md` files linking to each other preserving relative structure to one another. The following two checks should be sufficient on this repository to ensure integrity of the docs page:
- Existence of the `SUMMARY.md`
- All internal MD links are valid (this could be good? https://github.com/gaurav-nelson/github-action-markdown-link-check)

### PR DESCRIPTION
**Pull request by mahula**
### Issues
- fixes #6816 

### Todo
- Check every push
   - [x] have Markdown files been changed?
      - [x] do all Markdown files listed in SUMMARY.md exist (see [output, if files listed are not found](https://github.com/Ocelot-Social-Community/Ocelot-Social/actions/runs/6789326690/job/18456263956))?
      - [x] do all links in all Markdown files work (see [output, if links in Markdown files do not work](https://github.com/Ocelot-Social-Community/Ocelot-Social/actions/runs/6789326690/job/18456263755))?

### CODE REVIEW

File: .github/workflows/check-documentation.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+name: ocelot.social check documentation
+
+on:
+  push:
+    branches:
+      - 6816-refactor-docsocelotsocial-ensure-vuepress-does-not-break
```
Comment (ulfgebhardt) [inline .github/workflows/check-documentation.yml:6]:
```suggestion
on: push
```

File: .github/workflows/check-documentation.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  check-summary:
+    name: Check documentation summary
+    if: needs.files-changed.outputs.markdown-files == 'true'
+    needs: files-changed
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@master
+
+      - name: Check, if all .md files listed in SUMMARY.dm exist in current branch
```
Comment (ulfgebhardt) [inline .github/workflows/check-documentation.yml:35]:
```suggestion
      - name: Check, if all .md files listed in SUMMARY.md exist in current branch
```

File: .github/workflows/check-documentation.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+          if [ ${#missing_files[@]} -eq 0 ]; then
+            echo "All .md files listed in SUMMARY.dm exist."
+          else
+            echo "These files listed in SUMMARY.dm do not exist:"
+            for missing_file in "${missing_files[@]}"; do
+              echo "$missing_file"
+            done
+            exit 1
+          fi
```
Comment (ulfgebhardt) [inline .github/workflows/check-documentation.yml:59]: Why is this needed?
Is the check below `Check Markdown links` not sufficient?

File: .github/workflows/check-documentation.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      uses: actions/checkout@master
+    - name: Check Markdown Links
+      uses: gaurav-nelson/github-action-markdown-link-check@master
+      with:
+        use-quiet-mode: 'yes'
+        use-verbose-mode: 'no'
+        # at any .md file change take the chance to check the links in all .md files
+        check-modified-files-only: 'no'
+        config-file: '.github/workflows/mlc_config.json'
+        base-branch: 'trigger-docs.ocelot.social-deployment-by-markdown-file-changes-in-master-branch' # set to branch of current PR's branch
```
Comment (ulfgebhardt) [inline .github/workflows/check-documentation.yml:77]:
```suggestion
```
As I understand this, this is the base branch - so the branch to compare to - you should run with the default: `master`

File: .github/workflows/mlc_config.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  "retryOn429": true,
+  "retryCount": 2,
+  "fallbackRetryDelay": "30s",
+  "aliveStatusCodes": [200, 206],
+  "ignorePatterns": [
+    {
+      "pattern": "^(https:\/\/github.com\/Ocelot-Social-Community\/Ocelot-Social\/pull\/|http:\/\/localhost)"
+    }
+  ]
+}
```
Comment (ulfgebhardt) [inline .github/workflows/mlc_config.json:12]: I am against the use of this file and would like to use the disable syntax:
```
<!-- markdown-link-check-disable -->
## Section

Disable link checking in this section. Ignore this [Bad Link](https://exampleexample.cox)
<!-- markdown-link-check-enable -->
```

File: .github/workflows/check-documentation.yml (hunk 3 above)
Comment (mahula) [reply .github/workflows/check-documentation.yml:59]: Thanx for the hint.
Right, this job is redundant, since it is part of `check-markdown-links` already.

File: .github/workflows/check-documentation.yml (hunk 4 above)
Comment (mahula) [reply .github/workflows/check-documentation.yml]: Right.
Resolved in e58b943d92fb8ac8a0ba9db384eee44c37d7468a.

File: .github/workflows/check-documentation.yml (hunk 1 above)
Comment (mahula) [reply .github/workflows/check-documentation.yml:6]: Of course, this will be the last commit for this PR.

File: .github/workflows/mlc_config.json (hunk 5 above)
Comment (ulfgebhardt) [reply .github/workflows/mlc_config.json:12]: Ok i was mistaken that this file resides in the main directory and therefore the connection to the workflow was not imminent visible to me. After checking again I see its within the workflow folder 👍🏻 !

**Review by Elweyn — APPROVED** [review]
